src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs

```diff
@@ -2125,6 +2125,200 @@ private static byte[] PreperateResponseWithRedirect(byte[] location)
+
+        [ConditionalFact(nameof(SupportsAlpn))]
+        public async Task Http2_MultipleConnectionsEnabled_ConnectionLimitNotReached_ConcurrentRequestsSuccessfullyHandled()
+        {
+            const int maxHttp2ConnectionsPerServer = 3;
+            const int maxConcurrentStreams = 2;
+            using (Http2LoopbackServer server = Http2LoopbackServer.CreateServer())
+            using (SocketsHttpHandler handler = new SocketsHttpHandler {
+                MaxHttp2ConnectionsPerServer = maxHttp2ConnectionsPerServer,
+                SslOptions = { RemoteCertificateValidationCallback = (object _, X509Certificate _, X509Chain _, SslPolicyErrors _) => true }
+            })
+            using (HttpClient client = CreateHttpClient(handler))
+            {
+                server.AllowMultipleConnections = true;
+                Task<HttpResponseMessage>[] sendTasks = new Task<HttpResponseMessage>[maxHttp2ConnectionsPerServer * maxConcurrentStreams];
+                List<Http2LoopbackConnection> connections = new List<Http2LoopbackConnection>();
+                for (int i = 0; i < sendTasks.Length; i++)
+                {
+                    sendTasks[i++] = client.GetAsync(server.Address);
+                    Http2LoopbackConnection connection = await server.EstablishConnectionAsync(new SettingsEntry { SettingId = SettingId.MaxConcurrentStreams, Value = maxConcurrentStreams });
+                    connections.Add(connection);
+                    sendTasks[i] = client.GetAsync(server.Address);
+                }
+
+                Task[] respondTasks = new Task[sendTasks.Length];
+                int respondTaskIndex = 0;
+                foreach (Http2LoopbackConnection connection in connections)
+                {
+                    for (int i = 0; i < maxConcurrentStreams; i++)
+                    {
+                        int streamId = await connection.ReadRequestHeaderAsync();
+                        respondTasks[respondTaskIndex++] = connection.SendDefaultResponseAsync(streamId);
+                    }
+                }
+
+                await Task.WhenAll(respondTasks).TimeoutAfter(TestHelper.PassingTestTimeoutMilliseconds);
+                await Task.WhenAll(sendTasks).TimeoutAfter(TestHelper.PassingTestTimeoutMilliseconds);
+
+                foreach (Task<HttpResponseMessage> sendTask in sendTasks)
+                {
+                    HttpResponseMessage response = await sendTask;
+                    Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                }
+            }
+        }
+
+        [ConditionalFact(nameof(SupportsAlpn))]
+        public async Task Http2_MultipleConnectionsEnabled_ConnectionLimitReached_ConcurrentRequestsQueued()
+        {
+            const int maxConcurrentStreams = 2;
+            const int totalRequestCount = 100;
+            using (Http2LoopbackServer server = Http2LoopbackServer.CreateServer())
```

Comment:
nit: to reduce indent count, consider for these using blocks:

```suggestion
            using Http2LoopbackServer server = Http2LoopbackServer.CreateServer();
```